elytron/src/main/java/org/wildfly/extension/elytron/CredentialStoreAliasDefinition.java

```diff
@@ -214,16 +170,17 @@ protected void performRuntime(OperationContext context, ModelNode operation, Res
          */
         @Override
         public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
-            transformOperationAddress(operation);
-            super.execute(context, operation);
+            ModelNode transformedOperation = transformOperation(context, operation);
+            if(!CASE_SENSITIVE.resolveModelAttribute(context, context.readResourceFromRoot(context.getCurrentAddress().getParent()).getModel()).asBoolean()) {
+                ALIAS_NAME.resolveModelAttribute(context, transformedOperation).asString();
```

Comment:
A reusable validator is nice, but this isn't a parameter, so the validator is providing a misleading message. I think we're better off providing a precise message, since this kind of constraint on resource names is not common and the reason why case matters isn't obvious.